### PR TITLE
Support sign up anonymously and edit username

### PR DESCRIPTION
**Pull request by cheungpat**
connects skygeario/skygear-SDK-JS#61
connects skygeario/skygear-SDK-JS#33

### CODE REVIEW

File: content/js/users.md [hunk 1]
```diff
@@ -140,6 +140,11 @@ While each of the sign-up functions is resolved with a user object,
 in most cases you need not deal with it because
 you can access the currently logged-in user using `skygear.currentUser`.
 
+If your app does not require the user to create an account before using your
+app, you can sign up the user anonymously. An user that is signed up
+anonymously does not have a username, email, nor a password. You can add a
+username, email and/or password to a user after signing up.
```
Comment (bensonby) [inline content/js/users.md]: Does it mean that a new user will be created for every anonymous signup? I am not sure about this feature - are there any implications on how to handle the anonymous user?

- each of these users still has their own user IDs
- can an anonymous user create records?
- does an anonymous user have a default role?

Comment (rickmak) [inline content/js/users.md]: - each of these users still has their own user IDs -> YES
- can an anonymous user create records? -> YES
- does an anonymous user have a default role? -> YES

anonymously is just normal user without username and password. And once you lost your access_token, you lost the account forever.

Comment (bensonby) [inline content/js/users.md]: Thanks rick. @cheungpat I think you can add more information about the behavior for an anonymous user including the points mentioned above

Comment (cheungpat) [inline content/js/users.md]: Not sure why people keep making distinction between users and users that are signed up anonymously. They are the same.

The function was added as a convenient function to create a user without username, email nor password. It essentially calls the underlying implementation that is also called by the `signup` function.

Additionally, the function was added to bring feature parity to the iOS SDKs which also have this method for a long time.

I thought this is pretty clear what it means to sign up anonymously: “An [sic] user that is signed up anonymously does not have a username, email, nor a password.”

**Pull request merged**